### ISSUE TITLE
Use different Optimizers for two branches of a network. Branches use each other to output results when finally calculating Loss

### ISSUE THREAD

**Issue by NiuYingchun** [tensorflow/tensorflow#56534]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Support

### Source

binary

### Tensorflow Version

tf1.11

### Custom Code

No

### Python version

3.5

### Current Behaviour?

```shell
I want to calculate the KLD loss using the output of the two branches, and there will always be an error where one of the results is empty.
```

### Standalone code to reproduce the issue

```shell
self.teacher_loss = self.get_teacher_loss()   #  branch 1 loss
 self.stu_loss = self.get_stu_loss()            # branch 2 loss
     
 with tf.variable_scope('teacher_optimizer'):   #optimizer 1
    ....
 with tf.variable_scope('stu_optimizer'):       #optimizer 2
    ...
```

### Relevant log output

_No response_</details>

**Comment by NiuYingchun** [comment]
Thank you! I try

**Comment by sushreebarsa** [comment]
@NiuYingchun Could you please let us know if you have tried to upgrade to the latest TF version?
Thank you!

**Comment by NiuYingchun** [comment]
> @NiuYingchun Could you please let us know if you have tried to upgrade to the latest TF version? Thank you!

Thank you, I'm trying. I haven't finished it yet

**Comment by NiuYingchun** [comment]
> @NiuYingchun Older versions of TF(1.x) are not actively supported. Please upgrade to latest TF version and let us know if the issue still persists in newer versions? Thank you!

Thank you! It worked in TF2

**Comment by sushreebarsa** [comment]
@NiuYingchun Glad it worked for you!
Thank you for the update. Could you please move this issue to closed status if it is resolved for you?
Thank you!
